### PR TITLE
[luci/export] Introduce encodeOpBufferPack4bit

### PR DESCRIPTION
**Pull request by seanshpark**
This will introdue encodeOpBufferPack4bit to replace two encodeOpBufferByDType implementations for S4 and U4 to reduce duplicate codes.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
for  #12793 , from draft #12743 (done local validation)

File: compiler/luci/export/src/CircleTensorExporter.cpp [hunk 1]
```diff
@@ -457,6 +409,31 @@ encodeOpBufferByDType<loco::DataType::STRING>(FlatBufferBuilder &builder, luci::
...
+  const uint32_t size = c->size<DT>();
+  const uint32_t raw_size = (size + 1) / 2;
+  std::vector<uint8_t> raw_data(raw_size);
+
+  for (uint32_t i = 0; i < raw_size; ++i)
+  {
+    uint32_t sidx = i * 2;
+    uint8_t data = static_cast<uint8_t>(c->at<DT>(sidx));
+    raw_data[i] = data & 0x0f;
+    sidx++;
```
Comment (seanshpark) [inline compiler/luci/export/src/CircleTensorExporter.cpp:425]: `sidx++` is extracted here for better code understanding

File: compiler/luci/export/src/CircleTensorExporter.cpp [hunk 2]
```diff
@@ -457,6 +409,31 @@ encodeOpBufferByDType<loco::DataType::STRING>(FlatBufferBuilder &builder, luci::
   return CreateBuffer(builder, array_offset);
 }
 
+template <loco::DataType DT>
+flatbuffers::Offset<circle::Buffer> encodeOpBufferPack4bit(FlatBufferBuilder &builder,
+                                                           luci::CircleConst *c)
+{
```
Comment (shs-park) [inline compiler/luci/export/src/CircleTensorExporter.cpp:415]: compressed into one function! 👍

**Review by shs-park — APPROVED** [review]
LGTM!
=)

**Review by Seunghui98 — APPROVED** [review]
LGTM :)

**Pull request merged**